src/cockpit/beiboot.py

```diff
@@ -309,8 +308,21 @@ async def run(args) -> None:
         bridge.write_control(message)
         bridge.ssh_peer.thaw_endpoint()
     except ferny.InteractionError as exc:
-        sys.exit(str(exc))
+        error = ferny.ssh_errors.get_exception_for_ssh_stderr(str(exc))
+        logger.debug("ferny.InteractionError: %s, interpreted as: %r", exc, error)
+        if isinstance(error, ferny.SshAuthenticationError):
+            problem = 'authentication-failed'
+        elif isinstance(error, ferny.SshHostKeyError):
+            problem = 'unknown-hostkey'
+        elif isinstance(error, OSError):
+            # usually DNS/socket errors
+            problem = 'unknown-host'
+        else:
+            problem = 'internal-error'
+        bridge.write_control(command='init', problem=problem, message=str(error))
```

Comment:
I assume that's enough to get Cockpit Client not to show the stderr output then?

Like, what happens if we leak some stderr and then send a proper init/problem?  Which one will get shown?  Can we test it?

Reply:
With this diff
```
--- src/cockpit/beiboot.py
+++ src/cockpit/beiboot.py
@@ -308,6 +308,8 @@ async def run(args) -> None:
         bridge.write_control(message)
         bridge.ssh_peer.thaw_endpoint()
     except ferny.InteractionError as exc:
+        import sys
+        sys.stderr.write("Uh, schlimme Dinge!\n")
         error = ferny.ssh_errors.get_exception_for_ssh_stderr(str(exc))
         logger.debug("ferny.InteractionError: %s, interpreted as: %r", exc, error)
         if isinstance(error, ferny.SshAuthenticationError):
```

... it will still show the "proper" error code and no stderr. But with

```diff
--- src/cockpit/beiboot.py
+++ src/cockpit/beiboot.py
@@ -308,6 +308,7 @@ async def run(args) -> None:
         bridge.write_control(message)
         bridge.ssh_peer.thaw_endpoint()
     except ferny.InteractionError as exc:
+        assert False, "Uh, schlimmere Dinge!"
         error = ferny.ssh_errors.get_exception_for_ssh_stderr(str(exc))
         logger.debug("ferny.InteractionError: %s, interpreted as: %r", exc, error)
         if isinstance(error, ferny.SshAuthenticationError):
```

you get the full glory:

![image](https://github.com/cockpit-project/cockpit/assets/200109/44598f48-6e8d-4929-8799-d6cf4ea070f5)

Reply:
TODO: fixed with and pre and expander.